core/src/main/java/org/ehcache/core/BaseCacheManager.java

```diff
@@ -0,0 +1,634 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core;
+
+import org.ehcache.Cache;
+import org.ehcache.CacheManager;
+import org.ehcache.Status;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.Configuration;
+import org.ehcache.config.ResourceType;
+import org.ehcache.config.RuntimeConfiguration;
+import org.ehcache.core.config.BaseCacheConfiguration;
+import org.ehcache.core.config.DefaultConfiguration;
+import org.ehcache.core.config.events.StoreEventSourceConfiguration;
+import org.ehcache.core.config.store.StoreConfigurationImpl;
+import org.ehcache.core.events.CacheEventDispatcher;
+import org.ehcache.core.events.CacheEventDispatcherFactory;
+import org.ehcache.core.events.CacheManagerListener;
+import org.ehcache.core.spi.LifeCycledAdapter;
+import org.ehcache.core.spi.ServiceLocator;
+import org.ehcache.core.spi.cache.Store;
+import org.ehcache.core.spi.service.CacheManagerProviderService;
+import org.ehcache.core.util.ClassLoading;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.CacheEventListenerConfiguration;
+import org.ehcache.event.CacheEventListenerProvider;
+import org.ehcache.spi.LifeCycled;
+import org.ehcache.spi.cache.ObservableCacheManager;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriterProvider;
+import org.ehcache.spi.loaderwriter.WriteBehindConfiguration;
+import org.ehcache.spi.loaderwriter.WriteBehindProvider;
+import org.ehcache.spi.serialization.SerializationProvider;
+import org.ehcache.spi.serialization.Serializer;
+import org.ehcache.spi.serialization.UnsupportedTypeException;
+import org.ehcache.spi.service.Service;
+import org.ehcache.spi.service.ServiceConfiguration;
+import org.ehcache.spi.service.ServiceCreationConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Support the common core of {@code CacheManager} functionality.
+ *
+ * @author Clifford W. Johnson
+ */
+public abstract class BaseCacheManager implements CacheManager, ObservableCacheManager {
+
+  private final DefaultConfiguration configuration;
+  private final ClassLoader cacheManagerClassLoader;
+  private final boolean useLoaderInAtomics;
+
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
```

Comment:
For my two cents, I think using `BaseCacheManager` (the "parent" class name) hides information potentially useful in problem source identification -- what we really need to know is the _actual_ class responsible for getting is here (`EhcacheManager` or `ClusteredCacheManager` in this case) ... simply using `BaseCacheManager` hides that so we have to rely on clues obtained from elsewhere to know which of the cache manager implementations is involved.